ptsa/data/TimeSeriesX.py

```diff
@@ -182,19 +182,24 @@ def to_hdf(self, filename, mode='w', compression=None,
 
     @classmethod
     def from_hdf(cls, filename, decode_string_arrays=True, encoding='utf-8'):
-        """
-        Deserialize from HDF5 using :mod:`h5py`.
+        """Deserialize from HDF5 using :mod:`h5py`.
         Parameters
         ----------
         filename : str
         decode_string_arrays: bool
             Arrays of bytes should be decoded into strings
         encoding: str
-            Encoding scheme to use for decoding
+            Encoding scheme to use for decoding. To read files saved
+            with earlier PTSA versions use 'legacy' (but note that
+            support for reading legacy files is deprecated, so do save
+            them out with to_hdf5 to update their format).
         Returns
         -------
         Deserialized instance
+
         """
+        if encoding.upper() == 'LEGACY':
```

Comment:
Why can't we detect an old format? It should be as easy as checking the ``ptsa_version`` attribute.